lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -416,6 +424,35 @@ public Optional<Location> createTemporaryDirectory(Location targetPath, String t
         return Optional.of(temporary);
     }
 
+    @Override
+    public Optional<UriLocation> preSignedUri(Location location, Duration ttl)
+            throws IOException
+    {
+        if (azureAuth instanceof AzureAuthOauth) {
+            throw new UnsupportedOperationException("Generating pre-signed URIs for Azure with OAuth2 authentication is not supported");
+        }
+        AzureLocation azureLocation = new AzureLocation(location);
+        BlobClient client = createBlobClient(azureLocation);
+        BlobSasPermission blobSasPermission = new BlobSasPermission()
+                .setReadPermission(true);
+
+        OffsetDateTime startTime = OffsetDateTime.now();
+        OffsetDateTime expiryTime = startTime.plus(ttl.toMillis(), ChronoUnit.MILLIS);
+
+        BlobServiceSasSignatureValues values = new BlobServiceSasSignatureValues(expiryTime, blobSasPermission)
+                .setStartTime(startTime)
+                .setExpiryTime(expiryTime)
+                .setPermissions(blobSasPermission);
+
+        createBlobContainerClient(azureLocation).generateSas(values);
+        try {
+            return Optional.of(new UriLocation(URI.create(client.getBlobUrl() + "?" + client.generateSas(values)), Map.of()));
```

Comment:
Can `client.getBlobUrl()` have params alrady?
Would be nicer to move from `client.getBlobUrl()` to builder and use it to add query param

Reply:
Nope. blobUrl is just an URL to the object (without any query params)